### ISSUE TITLE
shipit_static_analysis: Should not mention issues that are fixed in later patches on the same bug

### ISSUE THREAD

**Issue by mstange** [mozilla-releng/services#655]
This may already have been fixed, but I'm filing it anyway in case it hasn't.
I've observed this twice:

1. https://bugzilla.mozilla.org/show_bug.cgi?id=1402498#c12
Here the comment was pointing out a mistake in code that was removed in a later patch in the same patch series, and replaced with code that did not have the problem.

2. https://bugzilla.mozilla.org/show_bug.cgi?id=1403444#c161
This sounds similar but seems to have been a mistake anyway.

**Comment by jankeromnes** [comment]
Thanks for filing this issue!

1. This sounds tricky. Our bot currently analyzes each commit separately, so if one commit has a problem that a later commit fixes, the bot is going to complain about the first commit. Maybe we could change the bot's workflow to analyze the entire patch queue in one go?
2. I'll reply on the bug, but this particular review was accidentally triggered on closed bug while debugging the bot locally (so the bot analyzed the whole patch queue, and complained about the defect in the 4th commit, regardless of whether other commits fixed it, or whether the patch queue was already landed). Normally the bot should only analyze fresh pushes to MozReview.

**Comment by mstange** [comment]
Yes, ideally it would analyze the entire patch queue in one go, and then somehow figure out which of the patches introduces the code that the warning is about so that it can comment on the right patch. Or it could analyze both individual patches and the combined diff, then (somehow) match up the warnings in the two, and discard all warnings from individual patches that are not present in the combined diff.

**Comment by jankeromnes** [comment]
Hm, thanks a lot for your ideas. They sound pretty tricky to implement, but we should consider them, especially if there are many other cases where one patch introduces defects that another patch fixes.

Matching defects across multiple patches is non-trivial, and still in research I believe (because of corner cases where code gets refactored, or split up, or otherwise transformed in non-trivial ways).

Something that would make it a lot easier to analyze entire patch queues is the ability to post review comments on a global diff, regardless of how many patches are in the queue. MozReview doesn't offer this (even if it has a global diff view), but other platforms might, e.g. GitHub and maybe Phabricator. Maybe there is a way.

**Comment by marco-c** [comment]
This should be feasible with Phabricator.
